crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -186,56 +197,156 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
-        let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let internal_id = if self.cached_identifiers.main_device_internal_id.is_none() {
+            let url_get_id: String = self.end_point.get_url_for_get_id(device_id);
+            let req_builder =
+                |token: String| HttpRequestBuilder::get(&url_get_id).bearer_auth(token);
+
+            self.get_and_set_jwt_token().await?;
+            let request_builder =
+                req_builder(self.cached_identifiers.token.clone().unwrap_or_default());
+
+            let request = request_builder.build()?;
+
+            let res = match self.peers.http.await_response(request).await? {
+                Ok(response) => match response.status() {
+                    StatusCode::OK => Ok(Ok(response)),
+                    StatusCode::UNAUTHORIZED | StatusCode::FORBIDDEN => {
+                        let request_builder =
+                            req_builder(self.cached_identifiers.token.clone().unwrap_or_default());
```

Comment:
The very same token is used while the point is to retry with a fresh one.

Reply:
Thanks for pointing out., I did add `self.get_and_set_jwt_token().await?;` some how missed while refactoring,